File: 225.implement-stack-using-queues.java
/*
 * @lc app=leetcode id=225 lang=java
 *
 * [225] Implement Stack using Queues
 */

// @lc code=start

import java.util.ArrayList;

class MyStack {

    private int[] queue = new int[100];
    private int top = -1;

    public MyStack() {
        
    }
    
    public void push(int x) {
        top++;
        queue[top] = x;
    }
    
    public int pop() {
        int last = queue[top];
        top--;
        return last;
    }
    
    public int top() {
        return queue[top];
    }
    
    public boolean empty() {
        return top == -1;
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */
// @lc code=end

File: 225.implement-stack-using-queues.py
#
# @lc app=leetcode id=225 lang=python3
#
# [225] Implement Stack using Queues
#

# @lc code=start
class MyStack:

    def __init__(self):
        self.queue = []

    def push(self, x: int) -> None:
        self.queue.append(x)
        
    def pop(self) -> int:
        return self.queue.pop()

    def top(self) -> int:
        return self.queue[-1]
        

    def empty(self) -> bool:
        return not self.queue


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()
# @lc code=end

